### PR TITLE
updated user entity, user dto, removed user interface

### PR DESCRIPTION
**Pull request by Matt-Tan15**
Updated user entity to match our data schema, updated user dtos

### CODE REVIEW

**Review by raylu511 — APPROVED** [review]

File: src/dtos/users.dto.ts [hunk 1]
```diff
@@ -1,9 +1,20 @@
...
+export type LoginUserRequestDto = Pick<BaseUserDto, "email" | "password">
 
-  @IsString()
-  public password: string;
-}
+export type RegisterUserRequestDto = Omit<BaseUserDto, "id">
+
+export type AuthenticateUserRequestDto = Pick<BaseUserDto, "id">
+
+export type RegisterUserResponseDto = Omit<BaseUserDto, "password">
```
Comment (emir4ik-code) [inline src/dtos/users.dto.ts:18]: Is it necessary for all other types to be present here, if we only use [id] from this dto?

File: src/dtos/users.dto.ts [hunk 2]
```diff
@@ -1,9 +1,20 @@
...
-  @IsEmail()
-  public email: string;
+export type LoginUserRequestDto = Pick<BaseUserDto, "email" | "password">
 
-  @IsString()
-  public password: string;
-}
+export type RegisterUserRequestDto = Omit<BaseUserDto, "id">
+
+export type AuthenticateUserRequestDto = Pick<BaseUserDto, "id">
```
Comment (emir4ik-code) [inline src/dtos/users.dto.ts:16]: Do we need Authentication DTO at all? Are we going to store user ID in JWT payload?

File: src/dtos/users.dto.ts (hunk 1 above)
Comment (raylu511) [reply src/dtos/users.dto.ts:18]: I think we just leave it for now so we can quickly fix the typing in the auth controller/service

**Review by DLiSWE — APPROVED** [review]
Seems good to me.

**Pull request merged**